src/Avalonia.Base/PixelSize.cs

```diff
@@ -67,13 +68,38 @@ public PixelSize(int width, int height)
         /// </summary>
         /// <param name="s">The string.</param>
         /// <returns>The <see cref="PixelSize"/>.</returns>
+        /// <exception cref="FormatException"/>
         public static PixelSize Parse(string s)
         {
-            using (var tokenizer = new StringTokenizer(s, CultureInfo.InvariantCulture, exceptionMessage: "Invalid PixelSize."))
+            if (TryParse(s, out var result))
             {
-                return new PixelSize(
-                    tokenizer.ReadInt32(),
-                    tokenizer.ReadInt32());
+                return result;
+            }
+            throw new FormatException("Invalid PixelSize.");
+        }
+
+        /// <summary>
+        /// Try parsing <paramref name="source"/> as <see cref="PixelSize"/>.
+        /// </summary>
+        /// <param name="source">The <see cref="string"/> to parse.</param>
+        /// <param name="result">The result of parsing. if <paramref name="source"/> is not valid <paramref name="result"/> is <see cref="PixelSize.Empty"/> </param>
+        /// <returns><c>true</c> if <paramref name="source"/> is valid <see cref="PixelSize"/>, otherwise <c>false</c>.</returns>
+        public static bool TryParse([NotNullWhen(true)] string? source,
+            [MaybeNullWhen(false)] out PixelSize result)
```

Comment:
As mentioned before, the `[MaybeNullWhen(false)]` does absolutely nothing on the `PixelSize result` argument since it's a struct: please remove it.